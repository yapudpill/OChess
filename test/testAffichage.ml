open EntreeSortie.Affichage

(*** Test d'affichage (cf testAffichage.expected) ***)
let p =
  EntreeSortie.Fen.creer_partie_fen
    "r1bqk2r/pppp1ppp/2n2n2/2b1p3/2BPP3/2P2N2/PP3PPP/RNBQK2R w KQkq -"

let () =
  let e = p.echiquier in
  print_endline "~~~ Affichage sans couleurs ~~~\n";
  print_echiquier ~couleur:false e;
  Printf.printf "Trait : %s\n" (string_of_couleur p.trait);
  Printf.printf "Adversaire : %s\n\n" (string_of_couleur Noir);

  print_endline "~~~ Affichage avec couleurs sans argument optionnel ~~~\n";
  print_echiquier e;
  Printf.printf "Trait : %s\n" (string_of_couleur p.trait);
  Printf.printf "Adversaire : %s\n\n" (string_of_couleur Noir);

  print_endline "~~~ Affichage avec couleurs avec argument optionnel ~~~\n";
  print_echiquier ~couleur:true e;
  Printf.printf "Trait : %s\n" (string_of_couleur p.trait);
  Printf.printf "Adversaire : %s\n" (string_of_couleur Noir)
