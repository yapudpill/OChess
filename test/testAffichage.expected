~~~ Affichage sans couleurs ~~~

8  ♖  .  ♗  ♕  ♔  .  .  ♖ 
7  ♙  ♙  ♙  ♙  .  ♙  ♙  ♙ 
6  .  .  ♘  .  .  ♘  .  . 
5  .  .  ♗  .  ♙  .  .  . 
4  .  .  ♝  ♟  ♟  .  .  . 
3  .  .  ♟  .  .  ♞  .  . 
2  ♟  ♟  .  .  .  ♟  ♟  ♟ 
1  ♜  ♞  ♝  ♛  ♚  .  .  ♜ 
   A  B  C  D  E  F  G  H 
Trait : Blanc
Adversaire : Noir

~~~ Affichage avec couleurs sans argument optionnel ~~~

8 [106m [30m♜ [44m   [106m [30m♝ [44m [30m♛ [106m [30m♚ [44m   [106m   [44m [30m♜ [0m
7 [44m [30m♟ [106m [30m♟ [44m [30m♟ [106m [30m♟ [44m   [106m [30m♟ [44m [30m♟ [106m [30m♟ [0m
6 [106m   [44m   [106m [30m♞ [44m   [106m   [44m [30m♞ [106m   [44m   [0m
5 [44m   [106m   [44m [30m♝ [106m   [44m [30m♟ [106m   [44m   [106m   [0m
4 [106m   [44m   [106m [97m♝ [44m [97m♟ [106m [97m♟ [44m   [106m   [44m   [0m
3 [44m   [106m   [44m [97m♟ [106m   [44m   [106m [97m♞ [44m   [106m   [0m
2 [106m [97m♟ [44m [97m♟ [106m   [44m   [106m   [44m [97m♟ [106m [97m♟ [44m [97m♟ [0m
1 [44m [97m♜ [106m [97m♞ [44m [97m♝ [106m [97m♛ [44m [97m♚ [106m   [44m   [106m [97m♜ [0m
   A  B  C  D  E  F  G  H 
Trait : Blanc
Adversaire : Noir

~~~ Affichage avec couleurs avec argument optionnel ~~~

8 [106m [30m♜ [44m   [106m [30m♝ [44m [30m♛ [106m [30m♚ [44m   [106m   [44m [30m♜ [0m
7 [44m [30m♟ [106m [30m♟ [44m [30m♟ [106m [30m♟ [44m   [106m [30m♟ [44m [30m♟ [106m [30m♟ [0m
6 [106m   [44m   [106m [30m♞ [44m   [106m   [44m [30m♞ [106m   [44m   [0m
5 [44m   [106m   [44m [30m♝ [106m   [44m [30m♟ [106m   [44m   [106m   [0m
4 [106m   [44m   [106m [97m♝ [44m [97m♟ [106m [97m♟ [44m   [106m   [44m   [0m
3 [44m   [106m   [44m [97m♟ [106m   [44m   [106m [97m♞ [44m   [106m   [0m
2 [106m [97m♟ [44m [97m♟ [106m   [44m   [106m   [44m [97m♟ [106m [97m♟ [44m [97m♟ [0m
1 [44m [97m♜ [106m [97m♞ [44m [97m♝ [106m [97m♛ [44m [97m♚ [106m   [44m   [106m [97m♜ [0m
   A  B  C  D  E  F  G  H 
Trait : Blanc
Adversaire : Noir
